libs/expo/betterangels/src/lib/screens/Interactions/InteractionsFilters/AuthorsFilter.tsx

```diff
@@ -0,0 +1,176 @@
+  const { data, loading, error } = useInteractionAuthorsQuery({
+    variables: {
+      pagination: { limit: paginationLimit, offset: offset },
+      filters: {
+        search: filterSearch,
+      },
+    },
+  });
+
+  const handleOnDone = () => {
+    setFilters({ ...filters, authors: selected });
+    setIsModalVisible(false);
+  };
+
+  const handleCloseModal = () => {
+    setSelected(filters.authors);
+    setIsModalVisible(false);
+  };
+
+  const handleSelectButtonPress = () => {
+    setSelected(filters.authors);
+    setIsModalVisible(true);
+  };
+
+  const loadMoreInteractions = useCallback(() => {
+    if (!hasMore || loading) return;
+
+    setOffset((prevOffset) => prevOffset + paginationLimit);
+  }, [hasMore, loading]);
+
+  const handleScroll = (event: NativeSyntheticEvent<NativeScrollEvent>) => {
+    const { layoutMeasurement, contentOffset, contentSize } = event.nativeEvent;
+
+    const isCloseToBottom =
+      layoutMeasurement.height + contentOffset.y >= contentSize.height - 20;
+
+    if (isCloseToBottom) {
+      loadMoreInteractions();
+    }
+  };
+
+  const debounceFetch = useMemo(
+    () =>
+      debounce((text) => {
+        setFilterSearch(text);
+        setOffset(0);
+      }, 500),
+    []
+  );
+
+  const onSearch = (e: string) => {
+    setSearch(e);
+    debounceFetch(e);
+  };
+
+  useEffect(() => {
+    if (!data?.interactionAuthors) return;
+
+    const { totalCount, results } = data.interactionAuthors;
+
+    const filteredAuthors = results
+      .filter((item) => item.id !== user?.id && item.firstName)
+      .map((item) => ({
+        id: item.id,
+        label: `${item.firstName} ${item.lastName}`,
+      }));
+
+    if (offset === 0) {
+      setAuthorsState(filteredAuthors);
+    } else {
+      setAuthorsState((prevAuthors) => [...prevAuthors, ...filteredAuthors]);
+    }
+
+    setHasMore(offset + paginationLimit < totalCount);
+  }, [data, offset]);
+
+  if (error || !user) throw new Error('Something went wrong!');
+
+  return (
+    <View>
+      <SelectButton
+        defaultLabel="All Authors"
+        selected={filters.authors?.map((item) => item.label)}
+        onPress={handleSelectButtonPress}
+      />
+
+      <Modal
+        isModalVisible={isModalVisible}
+        closeModal={handleCloseModal}
+        closeButton
+        vertical
+      >
+        <View
+          style={{ paddingHorizontal: Spacings.md, flex: 1, gap: Spacings.lg }}
+        >
+          <ScrollView
+            onScroll={handleScroll}
+            scrollEventThrottle={200}
+            showsVerticalScrollIndicator={false}
+          >
+            <MultiSelect
+              filterPlaceholder="Search"
+              withFilter
+              title="Filter - Authors"
+              search={search}
+              onSearch={onSearch}
+              onChange={(e: { id: string; label: string }[]) => setSelected(e)}
+              options={[{ id: user.id, label: 'Me' }, ...authorsState]}
```

Comment:
@Davit-BetterAngels what about something like this? 
```suggestion
              options={search ? [{ id: user.id, label: 'Me' }] : [...authorsState]}
```

Reply:
actually... if we're always showing the list of authors, it would be 
```suggestion
              options={
                search === ''
                  ? [{ id: user.id, label: 'Me' }, ...authorsState]
                  : [...authorsState]
              }
```